Assets/Code/Scripts/LevelManagement/CheckpointController.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CheckpointController : MonoBehaviour
+{
+    private GameManager _gameManager;
+    /*
+     * CheckpointController checks if the player has collided with a checkpoint.
+     * If a new checkpoint is hit, then it'll be overwritten.
+     */
+    void Start()
+    {
+        _gameManager = GameObject.FindGameObjectWithTag("GameManager").GetComponent<GameManager>();
+    }
+    private void OnTriggerEnter(Collider otherObject)
+    {
+        if (otherObject.gameObject.name == "RealityPlayer")
```

Comment:
What do you think about tags comparison? I think that it is a better way to do the check because the tag can be assigned to a multiple objects

```suggestion
        if (otherObject.CompareTag("RealityPlayer"))
```

Reply:
Yes, I think tags are the best way to let various functions work. 

In fact, I had CompareTag before the name compare, simply because tags weren't very clear. We may need a more strict tag convention.

I'll revert to the CompareTag. Thanks!